ui/src/lib.rs

```diff
@@ -1312,9 +1312,11 @@ enum Interrupt {
 impl Interrupt {
     fn char_press(&self) -> Option<char> {
         if let Self::KeyPressed(event) = &self {
-            if let KeyCode::Char(c) = event.code {
-                if c.is_alphanumeric() || c == '\'' {
-                    return Some(c)
+            if matches!(event.kind, KeyEventKind::Press) {
+                if let KeyCode::Char(c) = event.code {
+                    if c.is_alphanumeric() || c == '\'' {
+                        return Some(c)
+                    }
                 }
```

Comment:
this is independent of the change above?

Reply:
yea independent change

Reply:
I'd like to merge this individually
can you split these changes into two, the spec/env changes needs some investigation